### PR TITLE
HMAN-39

### PR DESCRIPTION
**Pull request by padmapriyanalam**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/HMAN-39

### Change description ###

Check "Status Transition" for PUT/DELETE hearing request

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/hmc/domain/model/enums/DeleteHearingStatus.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package uk.gov.hmcts.reform.hmc.domain.model.enums;
+
+public enum DeleteHearingStatus {
+    HEARING_REQUESTED,
+    UPDATE_REQUESTED,
+    UPDATE_SUBMITTED,
+    AWAITING_LISTING,
+    LISTED;
+
+    public static boolean chekStatus(String status) {
```
Comment (danlysiak) [inline src/main/java/uk/gov/hmcts/reform/hmc/domain/model/enums/DeleteHearingStatus.java:10]: Typo in `chek`, but perhaps a more suitable name:
```suggestion
    public static boolean isValid(String status) {
```
Same comment applies to `UpdateHearingStatus`

File: src/main/java/uk/gov/hmcts/reform/hmc/domain/model/enums/DeleteHearingStatus.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        for (DeleteHearingStatus enumStatus : values()) {
+            if (enumStatus.name().equals(status)) {
+                return true;
+            }
+        }
+        return false;
```
Comment (danlysiak) [inline src/main/java/uk/gov/hmcts/reform/hmc/domain/model/enums/DeleteHearingStatus.java:16]:
```suggestion
        return Arrays.stream(values()).anyMatch(enumStatus -> enumStatus.name().equals(status));
```
Optional one-liner alternative 😄 

Same comment applies to `UpdateHearingStatus`

**Review by danlysiak — COMMENTED** [review]
One actionable comment in code. Additionally can we add integration tests for the controller to confirm it returns the 400 properly please.

File: src/main/java/uk/gov/hmcts/reform/hmc/domain/model/enums/DeleteHearingStatus.java (hunk 1 above)
Comment (padmapriyanalam) [reply src/main/java/uk/gov/hmcts/reform/hmc/domain/model/enums/DeleteHearingStatus.java]: changed

File: src/main/java/uk/gov/hmcts/reform/hmc/domain/model/enums/DeleteHearingStatus.java (hunk 2 above)
Comment (padmapriyanalam) [reply src/main/java/uk/gov/hmcts/reform/hmc/domain/model/enums/DeleteHearingStatus.java]: changed in both places

**Review by danlysiak — APPROVED** [review]
✅

**Pull request merged**